optimum/intel/openvino/modeling_diffusion.py

```diff
@@ -838,17 +846,23 @@ def reshape(
 
         if self.text_encoder is not None:
             self.text_encoder.model = self._reshape_text_encoder(
-                self.text_encoder.model, batch_size, self.tokenizer.model_max_length
+                self.text_encoder.model,
+                batch_size,
+                self.tokenizer.model_max_length if "Gemma" not in self.tokenizer.__class__.__name__ else -1,
             )
 
         if self.text_encoder_2 is not None:
             self.text_encoder_2.model = self._reshape_text_encoder(
-                self.text_encoder_2.model, batch_size, self.tokenizer_2.model_max_length
+                self.text_encoder_2.model,
+                batch_size,
+                self.tokenizer_2.model_max_length if "Gemma" not in self.tokenizer.__class__.__name__ else -1,
```

Comment:
the check should be `if "Gemma" not in self.tokenizer_2.__class__.__name__` ?
```suggestion
                getattr(self.tokenizer_2, "model_max_length", -1),
```

Reply:
it does not solve issue, because actually this field exists but equal to max int